sim/hunter/volley.go

```diff
@@ -0,0 +1,102 @@
+package hunter
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+func (hunter *Hunter) registerVolleySpell() {
+	ranks := 3
+
+	for i := ranks; i >= 0; i-- {
+		config := hunter.getVolleyConfig(i)
+
+		if config.RequiredLevel <= int(hunter.Level) {
+			hunter.Volley = hunter.GetOrRegisterSpell(config)
+			break
+		}
+	}
+}
+
+func (hunter *Hunter) getVolleyConfig(rank int) core.SpellConfig {
+	spellId := [4]int32{0, 1510, 14294, 14295}[rank]
+	baseDamage := [4]float64{0, 50, 65, 80}[rank]
+	manaCost := [4]float64{0, 350, 420, 490}[rank]
+	level := [4]int{0, 40, 50, 58}[rank]
+
+	hasImprovedVolley := hunter.HasRune(proto.HunterRune_RuneCloakImprovedVolley)
+
+	manaCostModifer := 1.0
+	manaCostModifer *= 1 - (0.02 * float64(hunter.Talents.Efficiency))
+
+	if hasImprovedVolley {
+		manaCostModifer *= 0.5
+	}
+
+
+	return core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: spellId},
+		SpellSchool: core.SpellSchoolArcane,
+		ProcMask:    core.ProcMaskSpellDamage,
+		Flags:       core.SpellFlagChanneled | core.SpellFlagAPL,
+
+		RequiredLevel: level,
+		Rank:          rank,
+
+		ManaCost: core.ManaCostOptions{
+			FlatCost: manaCost,
+			Multiplier: manaCostModifer,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    hunter.NewTimer(),
+				Duration: time.Second * 60,
+			},
+		},
+		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+			return !hunter.Unit.Moving
+		},
```

Comment:
Is this needed if we have the `core.SpellFlagChanneled` flag on it? If not then I think the core should handle this kind of condition